Library/Homebrew/rubocops/extend/formula_cop.rb

```diff
@@ -369,7 +369,14 @@ def problem(msg)
 
       def formula_class?(node)
         _, class_node, = *node
-        class_node && string_content(class_node) == "Formula"
+        class_names = %w[
+          Formula
+          GithubGistFormula
+          ScriptFileFormula
+          AmazonWebServicesFormula
+        ]
+
+        class_node && class_names.index(string_content(class_node))
```

Comment:
If you can get the name or `Class` object of a class you could recurse through the inheritance hierarchy. That's maybe unnecessary, though but just an idea. I'm fine with this approach.